app_check/src/android/app_check_android.cc

```diff
@@ -294,6 +349,14 @@ AppCheckInternal::~AppCheckInternal() {
   JNIEnv* env = app_->GetJNIEnv();
   app_ = nullptr;
 
+  if (j_app_check_listener_ != nullptr) {
```

Comment:
Should probably also clear out listeners_

Reply:
Done. I just added a call to listeners_.clear();
I don't call delete on the listener pointers as they are allocated by non-firebase code.